### PR TITLE
Reinstall

### PR DESCRIPTION
**Pull request by stepasm**

### CODE REVIEW

File: dnf-docker-test/features/reinstall-1.feature [hunk 1]
```diff
@@ -0,0 +1,25 @@
+Feature: reinstall
```
Comment (ignatenkobrain) [inline dnf-docker-test/features/reinstall-1.feature]: needs more description ;)

File: dnf-docker-test/features/reinstall-1.feature [hunk 2]
```diff
@@ -1,4 +1,4 @@
-Feature: reinstall
+Feature: Reinstall won't break dependencies
```
Comment (ignatenkobrain) [inline dnf-docker-test/features/reinstall-1.feature]: that's not true ;)

Something like `Re-install must not change "reason" of package which has been re-installed`

File: dnf-docker-test/features/reinstall-1.feature [hunk 3]
```diff
@@ -1,6 +1,8 @@
-Feature: Reinstall won't break dependencies
+Feature: Re-install must not change "reason" of package which has been 
+installed as dependency
```
Comment (ignatenkobrain) [inline dnf-docker-test/features/reinstall-1.feature]: newline is not supported by behave

File: dnf-docker-test/features/reinstall-1.feature [hunk 4]
```diff
@@ -1,6 +1,8 @@
-Feature: Reinstall won't break dependencies
+Feature: Re-install must not change "reason" of package which has been 
+installed as dependency 
 
-Scenario: Dnf installs foo, that requires foo-libs. After reinstall of foo-libs there will be still removed foo-libs when removed foo.
+Scenario: Dnf installs foo, that requires foo-libs. After reinstall of
+foo-libs there will be still removed foo-libs when removed foo.
```
Comment (ignatenkobrain) [inline dnf-docker-test/features/reinstall-1.feature]: newline is not supported by behave

**Comment by ignatenkobrain** [comment]
commit f974c3e21256cdf8b9ac675b8f1d4f0e42d2b70e
```
Author: Štěpán Smetana <ssmetana@redhat.com>
Date:   Wed Jul 27 11:45:15 2016 +0200

    Re-install must not change "reason" of package which has been installed as dependency
    
    References: https://github.com/rpm-software-management/dnf/pull/531
    References: https://bugzilla.redhat.com/show_bug.cgi?id=1343764
    Reviewed-by: Igor Gnatenko <ignatenko@redhat.com>
```

@stepasm do: `$ git config --global user.name Štěpán Smetana`

**Comment by stepasm** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;hOjCAr47&#x2F;461-dnf-reinstall-deletes-yumdb-entry"> dnf reinstall deletes yumdb entry</a>

**Pull request closed without merging**